app/jobs/withdraw_draft_applications_for_expired_vacancies_job.rb

```diff
@@ -0,0 +1,10 @@
+class WithdrawDraftApplicationsForExpiredVacanciesJob < ApplicationJob
+  queue_as :low
+
+  def perform
+    JobApplication.joins(:vacancy)
+                  .draft
+                  .merge(Vacancy.expired)
+                  .find_each { |ja| ja.update!(status: "withdrawn") }
```

Comment:
Suggestion:
 It seems we can avoid iterating/instantiating the objects and reduce the find & update to a single SQL query.
```suggestion
                  .update_all(status: "withdrawn")
```

Would execute this single query instead of individual updates for each Job Application:
```sql
ActiveRecord 
-- JobApplication Update All -- 
{ :sql => "UPDATE \"job_applications\" 
                SET \"status\" = $1 
                WHERE \"job_applications\".\"id\" IN 
                   (SELECT \"job_applications\".\"id\" 
                    FROM \"job_applications\" 
                    INNER JOIN \"vacancies\" 
                    ON \"vacancies\".\"id\" = \"job_applications\".\"vacancy_id\" 
                    WHERE \"job_applications\".\"status\" = $2 
                    AND \"vacancies\".\"status\" = $3 
                    AND (expires_at < '2024-09-11 14:39:46.593271'))",
 :binds => { :status => [ 5, 0, 0 ] }, :allocations => 5, :cached => nil }
```